lib/promql/parser.go

```diff
@@ -0,0 +1,1298 @@
+package promql
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// Panicf controls how this package reports a runtime error indicative of a
+// bug in the implementation.
+var Panicf func(format string, args ...interface{}) = func(format string, args ...interface{}) {
```

Comment:
This is a hook point to allow plugging in a Panicf function that does not rely on the `logger` package. By default it will do a simple panic, but the `vmselect` module plugs in `logger.Panicf` instead.